sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_response_handlers.py

```diff
@@ -59,8 +59,31 @@ def order_results(response, combined):
 
 
 def prepare_result(func):
+    def _get_error_code_and_message(error):
+        if hasattr(error.error, 'innererror') and error.error.innererror:
+            return error.error.innererror.code, error.error.innererror.message
+        return error.error.code, error.error.message
+
+    def _deal_with_too_many_documents(response, obj):
+        # special case for now if there are too many documents in the request
+        # they may change id to empty string, but currently it is "All"
+        too_many_documents_errors = [
+            error for error in obj.errors
+            if error.id == "All" and _get_error_code_and_message(error)[0] == "InvalidDocumentBatch"
```

Comment:
if error.error.code is returned from the above function then the code will be "InvalidArgument" basing it off the example we saw from the service team. Do you think we need this extra check?

Also sorry just thinking some more (i'm not sure what would happen in this case), but do we need to validate user input to make sure they don't pass in a "" as the id? This might already not be allowed and throw an exception, I'm just not sure.

Reply:
Yeah that makes sense, I'll remove the check for InvalidDocumentBatch.

It doesn't look like we're validating TextDocumentInput's ids: https://github.com/Azure/azure-sdk-for-python/blob/b835a91621decbc51007cb92e5151f5f6563e120/sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_request_handlers.py#L52. I'll add a check for "All" right now, and when the service team changes it to "", I'll change the check to that